src/core/shape_inference/include/irdft_shape_inference.hpp

```diff
@@ -0,0 +1,134 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+#include <openvino/op/irdft.hpp>
+
+#include "openvino/core/axis_vector.hpp"
+#include "utils.hpp"
+
+template <class T>
+void irdft_shape_infer(const ov::op::v9::IRDFT* op,
+                      const std::vector<T>& input_shapes,
+                      std::vector<T>& output_shapes,
+                      const std::map<size_t, std::shared_ptr<ngraph::runtime::HostTensor>>& constant_data = {}) {
+    using DimType = typename std::iterator_traits<typename T::iterator>::value_type;
```

Comment:
@jane-intel Do we need to shape inference implementation in this PR?

Reply:
We cannot implement the nGraph shell without shape inference implementation.